Jenkinsfile-check-versions

```diff
@@ -0,0 +1,56 @@
+library("tdr-jenkinslib")
+
+pipeline {
+  agent {
+    label "master"
+  }
+  parameters {
+    choice(name: "STAGE", choices: ["intg", "staging", "prod"], description: "The stage you are deploying Terraform changes to")
+  }
+  stages {
+    stage("Get version") {
+      steps {
+        script {
+          sh "docker pull jenkins/jenkins:alpine"
+          def latestVersion = sh(script: """docker inspect jenkins/jenkins:alpine | jq -r '.[0].ContainerConfig.Env[]' | grep JENKINS_VERSION | awk -F "=" '{print \$2}'""", returnStdout: true).trim()
+          def currentVersion = Jenkins.instance.getVersion().toString().trim()
+          if(currentVersion != latestVersion) {
+            slackSend color: "good", message: """Version ${latestVersion} is available for Jenkins
+See the developer manual for a guide on updating it
+https://github.com/nationalarchives/tdr-dev-documentation/blob/master/manual/update-jenkins.md#update-the-jenkins-application-version""", channel: "#da-tdr"
+          }
+        }
+      }
+    }
+    stage('Run Terraform build') {
+      agent {
+        ecs {
+          inheritFrom 'aws'
+          taskrole "arn:aws:iam::${env.MANAGEMENT_ACCOUNT}:role/TDRJenkinsCheckAmiRole"
+        }
+      }
+      stages {
+        stage('Check for ami updates') {
+          steps {
+            script {
+              def currentAmiId = sh(script: """aws s3 cp s3://tdr-terraform-state-jenkins/jenkins-terraform-state - | jq -r '.resources[] | select(.module=="module.jenkins_ami") | .instances[0].attributes.source_ami_id'""", returnStdout: true).trim()
+              def mostRecentAmiId = sh(script: """aws ec2 describe-images --owners amazon --filters "Name=name,Values=amzn2-ami-ecs-hvm-2.0.*-x86_64-ebs" --query 'sort_by(Images, &CreationDate)[-1].ImageId' | jq -r""", returnStdout: true).trim()
+              if(currentAmiId != mostRecentAmiId) {
+                slackSend color: "good", message: """A new ami ${mostRecentAmiId} is available for Jenkins
+See the developer manual for a guide on updating it
+https://github.com/nationalarchives/tdr-dev-documentation/blob/master/manual/update-jenkins.md#update-the-ec2-instance
+                  """, channel: "#da-tdr"
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  post {
+    always {
+      echo 'Deleting Jenkins workspace...'
+      deleteDir()
```

Comment:
Just out of curiosity, why does this build need to delete the workspace? I don't remember seeing it in most of the other builds.

Reply:
I don't remember either. I copied this off one of the terraform ones so maybe there was a good reason there. I don't think it's needed her either so I deleted it.